### ISSUE TITLE
Notification frozen on the screen

### ISSUE THREAD

**Issue by trolliama** [gnome-pomodoro/gnome-pomodoro#691]
## Issue Summary

Sometimes the notifications freezes on my screen. I cant remove it... I pass the mouse through it and dont go away, just stays there. I need to logout to remove them, not even locking the screen fix the bug

## Steps to Reproduce

1. Just wait for a notification to appear, eventually will get stuck

## Context (logs, error output, screencast, OS, Desktop  etc)

[Screencast from 2024-03-14 19-25-55.webm](https://github.com/gnome-pomodoro/gnome-pomodoro/assets/21108976/87c3ca2e-bad1-475f-b3aa-7ee4ac9e5093)
here.)

## Other information

I am using ubuntu 23.04

**Comment by kamilprusko** [comment]
Hi!

Are there any errors/warnings for gnome-shell at those times? I recommend using GNOME Logs app.

**Comment by trolliama** [comment]
I have this app installed and I am going to pay attention for those logs from now on. But how do identify the logs related to this issue?

Also, I found out a work around for now. I just lock the screen and click the square button (stop button) on the notification

**Comment by kamilprusko** [comment]
Messages with keywords: "IdleMonitor", "notification" or "pomodoro" may be relevant. Browsing history should be enough, no need to reproduce the issue... If there are no logs, it should be assumed that this issue doesn't leave a trace.

**Comment by trolliama** [comment]
Nothing related to pomodoro

**Comment by jcrd** [comment]
I am occasionally experiencing a similar issue. The notification seems to freeze during its dismissal animation.

![image](https://github.com/gnome-pomodoro/gnome-pomodoro/assets/44040597/f9124a49-f986-4a91-8286-baa3cc97357a)

This is more than a visual artifact. It blocks mouse input to windows below it such as browser tabs.
No log results for the suggested keywords.

System:
- Fedora Workstation 39
- Gnome 45.5 on Wayland
- Pomodoro 0.24.0

**Comment by porchy21** [comment]
@kamilprusko are you resolve this problem? I'm facing the same problem. 

System:

- Nobara Linux 40 (like Fedora 40)
- GNOME 46.6
- Pomodoro 0.26.0

For some reason, the notification freezes only occasionally after exiting the waiting mode.

**Comment by kamilprusko** [comment]
I've seen this happening with other notifications. I might be wrong, but the issue seems more likely to be on the GNOME Shell side.

**Comment by kamilprusko** [comment]
@Igetin it's worth checking gnome-shell log:
```
journalctl --user --since=yesterday /usr/bin/gnome-shell
```

Last time I experienced it looked like an issue caused by an MPRIS notification

**Comment by cheddarsunrae** [comment]
Yes I had it hang Fedora 44 hard today. Had to Alt-F2 and open console then pkill - which in turn nuked my whole gnome session. Not pretty. Too bad because I loved this, but I cannot use it again until this is addressed.

**Comment by kamilprusko** [comment]
There is a patch in gnome-shell that landed in gnome-shell 49.rc that matches this issue exactly:

```
commit 93a79cf61b2b404357e6f637cc9e7f324fb27a69
Author: everyx <lunt.luo@gmail.com>
Date:   Sat Sep 13 12:01:17 2025 +0800

    messageTray: Fix race condition causing stuck notification banners
    
    A race condition can occur where a notification banner gets stuck on screen
    and never disappears.
    
    This happens when a new notification needs to be shown while an old one
    is in the middle of its hiding animation. The process to show the new
    notification calls `remove_all_transitions()`, which forcefully stops
    the hiding animation of the old one.
    
    The cleanup logic, which destroys the old banner actor, was located in
    the `onComplete` handler of the animation. Since the animation was
    interrupted and not completed, this handler was never called, leaving a
    "zombie" banner on screen.
    
    To fix this, the call to the cleanup function is moved from the
    `onComplete` handler to an `onStopped` handler. This handler is
    guaranteed to be called whenever an animation stops, regardless of
    whether it finished successfully or was interrupted. This ensures the
    banner is always cleaned up correctly.
    
    Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/6006
    Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3861>
```

I've stress tested the notifications and found another issue. Updating notification while it gets hidden locks up the notifications - there are no artifacts, just no new notifications appear. The fix for this is in the extension repo [96c0815331be382e2b6ac6225f02e25eeba02e27](https://github.com/focustimerhq/gnome-shell-extension-focus-timer/commit/96c0815331be382e2b6ac6225f02e25eeba02e27).

I did not experience this issue after the gnome-shell fix landed. The second issue with the extension is harder to reproduce. 

Will test this over the weekend...

**Comment by cheddarsunrae** [comment]
It happened again today but was easier to kill. Process "take a break" is showing but allows me to click the timer to start a new work session - everything else is locked out. ctrl-alt-f2 saves you

**Comment by kamilprusko** [comment]
@cheddarsunrae Check gnome-shell logs https://github.com/focustimerhq/gnome-shell-extension-focus-timer/blob/main/CONTRIBUTING.md#troubleshooting

**Comment by kamilprusko** [comment]
This got fixed in gnome-shell 49.

The additional fix landed in [version 3](https://github.com/focustimerhq/gnome-shell-extension-focus-timer/releases/tag/3) of the extension. I only was able to reproduce this issue synthetically.

@cheddarsunrae what you describe - unusable gnome-shell must be for other reason. The issue here is that notifications gets jammed.
